src/app/shared/log-in/methods/log-in.methods-decorator.ts

```diff
@@ -1,6 +1,6 @@
 import { AuthMethodType } from '../../../core/auth/models/auth.method-type';
 
-const authMethodsMap = new Map();
+const authMethodsMap: Map<AuthMethodType, any> = new Map();
 
```

Comment:
Here you should add the `hasAuthMethodType` function that will return `true` if the map has that target key configured